cmd/oras/manifest/fetch.go

```diff
@@ -90,18 +93,21 @@ func fetchManifest(opts fetchOptions) error {
 		return err
 	}
 	if repo.Reference.Reference == "" {
-		return errors.NewErrInvalidReference(repo.Reference)
+		return oerrors.NewErrInvalidReference(repo.Reference)
 	}
 	repo.ManifestMediaTypes = opts.mediaTypes
 
 	// Fetch and output
 	var content []byte
 	if opts.fetchDescriptor {
-		content, err = cas.FetchDescriptor(ctx, repo, opts.targetRef, tagetPlatform)
+		content, err = cas.FetchDescriptor(ctx, repo, opts.targetRef, targetPlatform)
 	} else {
-		content, err = cas.FetchManifest(ctx, repo, opts.targetRef, tagetPlatform)
+		content, err = cas.FetchManifest(ctx, repo, opts.targetRef, targetPlatform)
 	}
 	if err != nil {
+		if errors.Is(err, errdef.ErrNotFound) {
+			return fmt.Errorf("no matching manifest was found in %s", opts.targetRef)
```

Comment:
How do we distinguish tag not found vs. manifest not found vs. platform not found? or do we have a more friendly error message?

Reply:
how about: `Error: no manifest with platform invalid was found` ?